test/test_fx.py

```diff
@@ -186,6 +189,142 @@ def test_resnet(self):
         assert (a - d).abs().max() < 2
         assert torch.allclose(d, e)
 
+    def test_native_callable(self):
+        # This test exercises the case where we use FX to translate from Python
+        # code to some native callable object
+        #
+        # For the purposes of testing, we use ElementwiseInterpreter defined
+        # in test_custom_class.cpp.
+        #
+        # We test that we can
+        # 1) Construct a native callable from FX IR
+        # 2) Construct a drop-in replacement module that delegates to the
+        #    native callable rather than the original code
+        # 3) Run both the original code and native callable wrapper with
+        #    equivalent results
+        # 4) TorchScript compile the native callable wrapper and confirm
+        #    equivalent results with the reference
+        # 5) TorchScript serialize and deserialize the native callable
+        #    and confirm equivalent results with the reference
+
+        # We use this simple Module as a reference computation
+        class MySimpleMod(torch.nn.Module):
+            def forward(self, x):
+                return 3.0 * x + x
+
+        msm = MySimpleMod()
+
+        # This is what a lowering pass might look like: a function that takes
+        # a valid nn.Module, symbolically traces it, lowers the Module to some
+        # representation, and wraps that representation up into another
+        # nn.Module instance that handles dispatch to the compiled/lowered code.
+        def lower_to_elementwise_interpreter(orig_mod : torch.nn.Module) -> torch.nn.Module:
+            # ===== Stage 1: Symbolic trace the module =====
+            mod = symbolic_trace(orig_mod)
+
+            # ===== Stage 2: Lower GraphModule representation to the C++
+            #       interpreter's instruction format ======
+            instructions = []
+            constant_idx = 0
+            constants = {}
+            fn_input_names = []
+
+            target_to_name = {
+                operator.add : "add",
+                operator.mul : "mul"
+            }
+
+            # For each instruction, create a triple
+            # (instruction_name : str, inputs : List[str], output : str)
+            # to feed into the C++ interpreter
+            for n in mod.graph.nodes:
+                target, args, out_name = n.target, n.args, n.name
+                assert len(n.kwargs) == 0, "kwargs currently not supported"
+
+                if n.op == 'placeholder':
+                    # Placeholders specify function argument names. Save these
+                    # for later when we generate the wrapper GraphModule
+                    fn_input_names.append(target)
+                elif n.op == 'call_function':
+                    assert target in target_to_name, "Unsupported call target " + target
+                    arg_names = []
+                    for arg in args:
```

Comment:
If this is a demo of how to use fx, then this should probably call `map_arg` eventhough it really wouldn't support anything that does. Maybe it should also make it more clear that it is not supporting certain features like kwargs.